### ISSUE TITLE
Nest can't resolve dependencies in the forRootAsync inject array

### ISSUE THREAD

**Issue by GlCap** [MichalLytek/typegraphql-nestjs#9]
Hello. I'm using the new forRootAsync method to inject some modules, but it's now working as intended.

``` yaml
[Nest] 50966   - 09/28/2020, 7:25:27 PM   [NestFactory] Starting Nest application...
[Nest] 50966   - 09/28/2020, 7:25:27 PM   [InstanceLoader] AppModule dependencies initialized +42ms
[Nest] 50966   - 09/28/2020, 7:25:27 PM   [ExceptionHandler] Nest can't resolve dependencies of the TYPEGRAPHQL_ROOT_MODULE_OPTIONS (PrismaService, ?). Please make sure that the argument ConfigService at index [1] is available in the GraphQLModule context.

Potential solutions:
- If ConfigService is a provider, is it part of the current GraphQLModule?
- If ConfigService is exported from a separate @Module, is that module imported within GraphQLModule?
  @Module({
    imports: [ /* the Module containing ConfigService */ ]
  })
 +1ms
Error: Nest can't resolve dependencies of the TYPEGRAPHQL_ROOT_MODULE_OPTIONS (PrismaService, ?). Please make sure that the argument ConfigService at index [1] is available in the GraphQLModule context.

Potential solutions:
- If ConfigService is a provider, is it part of the current GraphQLModule?
- If ConfigService is exported from a separate @Module, is that module imported within GraphQLModule?
  @Module({
    imports: [ /* the Module containing ConfigService */ ]
  })

    at Injector.lookupComponentInParentModules (/home/app/node_modules/@nestjs/core/injector/injector.js:192:19)
    at async Injector.resolveComponentInstance (/home/app/node_modules/@nestjs/core/injector/injector.js:148:33)
    at async resolveParam (/home/app/node_modules/@nestjs/core/injector/injector.js:102:38)
    at async Promise.all (index 1)
    at async Injector.resolveConstructorParams (/home/app/node_modules/@nestjs/core/injector/injector.js:117:27)
    at async Injector.loadInstance (/home/app/node_modules/@nestjs/core/injector/injector.js:81:9)
    at async Injector.loadProvider (/home/app/node_modules/@nestjs/core/injector/injector.js:38:9)
    at async Promise.all (index 15)
    at async InstanceLoader.createInstancesOfProviders (/home/app/node_modules/@nestjs/core/injector/instance-loader.js:43:9)
    at async /home/app/node_modules/@nestjs/core/injector/instance-loader.js:28:13
```
the PrismaService works by making the PrismaModule global, this is my AppModule:
```javascript 

@Module({
  imports: [
    ConfigModule.forRoot(),
    ConfigModule.forFeature(commonConfig),
    PrismaModule,
    TypeGraphQLModule.forRootAsync({
      inject: [PrismaService, ConfigService],
      useFactory: (prisma: PrismaService, config: ConfigService) => ({
        authChecker: typegraphQLAuthChecker,
        context: graphqlContext(prisma),
        emitSchemaFile: config.get('NODE_ENV') === 'development',
      }),
    }),
  ],
})
export default class AppModule {}
```

seems like the underlying GraphQLModules isn't receiving the injected modules, but I'm not a big expert of Nest's inner workings.

Am I missing something? 
Thank you for your attention

**Comment by MichalLytek** [comment]
`TypeGraphQLModule.forRootAsync` method also accepts `imports` property with an array, could you try that?

```ts
TypeGraphQLModule.forRootAsync({
  imports: [PrismaModule, ConfigModule.forFeature(commonConfig)],
  inject: [PrismaService, ConfigService],
  useFactory: (prisma: PrismaService, config: ConfigService) => ({
    // ...
  }),
}),
```

I'm also not a NestJS expert, maybe @codejamninja could help us 😉

**Comment by codejamninja** [comment]
@GlCap here is an example where I have it working. I am using my fork of `typegraphql-nestjs`, but since #8 was just accepted, you should theoretically be able to use the official module. I also make a yeoman generator that sets up nestjs with prisma.

https://github.com/codejamninja/generator-nestjs-prisma

**Comment by GlCap** [comment]
@codejamninja  I was following your examples all along but I missed your final exports array as I didn't thought it could be related to my issue.

Anyway it seems to be able to inject only global scoped modules, either by marking the injected module with the Global decorator or exporting them from the module that is currently importing the TypeGraphQLModule (the config module was working because in you example you have used the isGlobal option).

But it should be able to import module through the imports array,  that doesn't works since the underlying GraphQLModule is not receiving it during instantiation, that is easily fixed with a one line of code.
```javascript
  static forRootAsync(
    asyncOptions: TypeGraphQLRootModuleAsyncOptions,
  ): DynamicModule {
     const dynamicGraphQLModule = GraphQLModule.forRootAsync({
 +     imports: asyncOptions.imports,
       useClass: TypeGraphQLOptionsFactory,
     });
...

```

**Comment by codejamninja** [comment]
Awesome, great find. Could you submit a PR?

**Comment by MichalLytek** [comment]
Fixed in 3c36969 🔒

**Comment by codejamninja** [comment]
Nice

**Comment by MichalLytek** [comment]
Sorry for delay, released in `v0.2.5` 🚀
